src/mem/threadalloc.h

```diff
@@ -167,10 +182,15 @@ namespace snmalloc
 #  ifdef USE_MALLOC
       return get_reference();
 #  else
-      auto alloc = get_reference();
+      auto*& alloc = get_reference();
       if (unlikely(needs_initialisation(alloc)))
```

Comment:
Adding an `&& !has_run` avoids a useless aquire/release in `init_thread_allocator`